autoscale/src/main/java/com/sequenceiq/periscope/service/ScalingActivityService.java

```diff
@@ -190,14 +205,22 @@ public Set<Long> findAllInStatusesThatStartedBefore(Collection<ActivityStatus> s
     public void deleteScalingActivityByIds(Set<Long> activityIds) {
         LOGGER.info("Deleting {} scalingActivities by their Ids", activityIds.size());
         scalingActivityRepository.deleteAllById(activityIds);
+        calculateUnprocessedScalingActivityMetrics();
     }
 
     public void deleteScalingActivityForCluster(Long clusterId) {
         LOGGER.info("Deleting all scaling activity for clusterId: {}", clusterId);
         scalingActivityRepository.deleteAllByCluster(clusterId);
     }
 
-    public void save(ScalingActivity scalingActivity) {
-        scalingActivityRepository.save(scalingActivity);
+    public ScalingActivity save(ScalingActivity scalingActivity) {
+        return scalingActivityRepository.save(scalingActivity);
+    }
+
+    private void calculateUnprocessedScalingActivityMetrics() {
+        // TODO: Should we fetch a broader collection of scaling events?
+        Collection<Long> activityIds = findAllInStatusesThatStartedBefore(ActivityStatus.getStatusesApplicableForCleanup(),
```

Comment:
Getting this as a Collection, and then counting it is not great. This needs to be a count at the SQL level itself.

Which brings up another aspect. The CleanupMonitor is going to perform the exact same query. So if, for some reason, the cleanup monitor has not run in some time - it could end up pulling in enough events to overwhelm the system. @aakashshankar  - can you please file a separate jira to track this, where the CleanupMonitor limits the number of events it tries to clean up in a single run ?

Reply:
Modified to count at the SQL level, also filed https://jira.cloudera.com/browse/CB-21498 to limit the events cleaned up by CleanupMonitor